app.rb

```diff
@@ -0,0 +1,112 @@
+require_relative './book'
+require_relative './person'
+require_relative './rental'
+require_relative './student'
+require_relative './teacher'
+
+# class to have basic UI
+class App
+  def list_books
+    if Book.all.empty?
+      puts 'There are no books in the database yet!'
+    else
+      Book.all.each do |book|
+        puts "Title: #{book.title}, Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if Student.all.empty? && Teacher.all.empty?
+      puts 'There are no people in the database yet!'
+    else
+      Student.all.each do |student|
+        puts "[Student] Name: #{student.name}, ID: #{student.id}, Age: #{student.age}"
+      end
+      Teacher.all.each do |teacher|
+        puts "[Teacher] Name: #{teacher.name}, ID: #{teacher.id}, Age: #{teacher.age}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a Teacher (2)? [Input the number]: '
+    select = gets.chomp.to_i
+    if [1, 2].include?(select)
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      case select
+      when 1
+        print 'Classroom: '
+        classroom = gets.chomp
+        print 'Has parent permission? [Y/N]: '
+        permission = gets.chomp
+        unless %w[Y N y n].include?(permission)
+          puts 'Invalid input, please write Y or N'
+          print 'Has parent permission? [Y/N]: '
+        end
+        Student.new(age, classroom, name: name, parent_permission: permission)
+        puts 'Student added the database successfully!'
+      when 2
+        print 'Specialization: '
+        specialization = gets.chomp
+        Teacher.new(age, specialization, name: name)
+        puts 'Teacher added to the database successfully!'
+      end
+    else
+      puts 'Invalid option, please type 1 or 2'
+      nil
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    Book.new(title, author)
+    puts 'Book added to the database successfully!'
+  end
```

Comment:
- [OPTIONAL] Make sure books are created in ascending order. This means the first book should have the first id and the last should have the highest book id

![Screenshot from 2023-01-13 04-54-40](https://user-images.githubusercontent.com/79058364/212227733-c0f035a8-78fd-4499-9834-be6cfccf21dd.png)

Reply:
thank you will fix that too!